Address linting issues

## Description
Currently, the Go linting worflows [fail](https://github.com/ZenPrivacy/zen-core/actions/runs/18815063290/job/53682099420) with the following error:
```
Error: filter/filter.go:76:6: exported: type name will be used as filter.FilterListType by other packages, and that stutters; consider calling this ListType (revive)
  type FilterListType string
       ^
Error: filter/filter.go:82:6: exported: type name will be used as filter.FilterList by other packages, and that stutters; consider calling this List (revive)
  type FilterList struct {
```

The struct and the type should be renamed to address the issue.


## Replication steps
1. Run `task lint-go` in the project root directory

## Acceptance criteria
- [ ] `golangci-lint run` executes with no errors

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,0 +1,1 @@
+*.go text eol=lf

--- a/filter/filter.go
+++ b/filter/filter.go
@@ -76,7 +76,7 @@ type whitelistSrv interface {
 type ListType string
 
 const (
-	FilterListTypeCustom ListType = "custom"
+	ListTypeCustom ListType = "custom"
 )
 
 type List struct {

